modules/porous_flow/include/materials/PorousFlowMaterialPermeabilityFromPorosity.h

```diff
@@ -0,0 +1,76 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#ifndef POROUSFLOWMATERIALPERMEABILITYFROMPOROSITY_H
+#define POROUSFLOWMATERIALPERMEABILITYFROMPOROSITY_H
+
+#include "DerivativeMaterialInterface.h"
+#include "Material.h"
+
+#include "PorousFlowDictator.h"
+
+//Forward Declarations
+class PorousFlowMaterialPermeabilityFromPorosity;
+
+template<>
+InputParameters validParams<PorousFlowMaterialPermeabilityFromPorosity>();
+
+/**
+ * Material designed to provide the permeability tensor
+ * which is a function of porosity
+ */
+class PorousFlowMaterialPermeabilityFromPorosity : public DerivativeMaterialInterface<Material>
+{
+public:
+  PorousFlowMaterialPermeabilityFromPorosity(const InputParameters & parameters);
+
+protected:
+  /// flags to indicate whether parameters have been set
+  bool _k0_set;
+  bool _phi0_set;
+  bool _f_set;
+  bool _d_set;
+  bool _k_anisotropy_set;
+  
+  /// parameters for poroperm equation
+  const Real _k0;
+  const Real _phi0;
```

Comment:
You probably want to comment these parameters.  The comment can be the same.  Eg

/// Permeability = porosity^_n / (1 - porosity)^_m , etc

They are for doxygen, so that someone will click on "_phi0" and hope to get an explanation of what it is.